azure/test/client.test.js

```diff
@@ -78,4 +75,101 @@ describe('Azure Log Monitoring', function() {
       assert.deepEqual(expectedMetadata, client.extractResourceId(record))
     });
   })
+
+  function fakeContext() {
+    // create a fake context object to pass into handleLogs
+    contextSpy = sinon.spy()
+    contextSpy.log = sinon.spy()
+    contextSpy.log.error = function(x) {} // do nothing
+    contextSpy.log.warn = function(x) {} // do nothing
+
+    return contextSpy
+  }
+
+  function testHandleLogs(logs, expected, isJson) {
+    // create a spy to mock the sender call
+   var handleJsonLogsSpy = sinon.spy();
+   var handleStringLogsSpy = sinon.spy();
+
+   sender = function(tagger){
+     if (tagger === client.addTagsToJsonLog) {
+       return handleJsonLogsSpy;
+      } else {
+       return handleStringLogsSpy;
+      }
+    }
+
+    client.handleLogs(sender, record, fakeContext());
+    if (isJson == true) {
+      sinon.assert.calledWith(handleJsonLogsSpy, expected)
+    } else {
+      sinon.assert.calledWith(handleStringLogsSpy, expected)
+
+    }
+  }
+
+  describe('#handleLogs', function() {
+    it('should handle string properly', function() {
+      record = 'hello'
+      expected = 'hello'
+      assert.equal(client.getLogFormat(record), constants.STRING)
+      testHandleLogs(record, expected, false)
+    });
+
+    it('should handle json-string properly', function() {
+      record = '{"hello": "there"}'
+      expected = {'hello': 'there'}
+      assert.equal(client.getLogFormat(record), constants.JSON_STRING)
+      testHandleLogs(record, expected, true)
+    });
+
+    it('should handle json-object properly', function() {
+      record = {'hello': 'there'}
+      expected = {'hello': 'there'}
+      assert.equal(client.getLogFormat(record), constants.JSON_OBJECT)
+      testHandleLogs(record, expected, true)
+    });
+
+    it('should handle string-array properly', function() {
+      record = ['one message', 'two message']
+      expected = 'one message'
```

Comment:
why do we only expect one record to be returned here for these tests with multiple records?